File: src/main/java/com/jss/jssbackend/modules/miscellaneous/service/PaymentTypeService.java
package com.jss.jssbackend.modules.miscellaneous.service;

import java.util.List;

import com.jss.jssbackend.modules.miscellaneous.model.PaymentType;

public interface PaymentTypeService {
    public List<PaymentType> getPaymentTypes();

    public PaymentType getPaymentType(Integer id);
}

File: client/src/app/modules/miscellaneous/services/payment.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { PaymentType } from '../../miscellaneous/model/PaymentType';

@Injectable({
  providedIn: 'root'
})
export class PaymentTypeService extends AppRestService<PaymentType>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getPaymentTypes() {
    return this.getListCached(new HttpParams(), "payment-types");
  }

  addOrUpdatePaymentType(paymentType: PaymentType) {
    this.clearListCache(new HttpParams(), "payment-types");
    return this.addOrUpdate(new HttpParams(), "payment-type", paymentType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}
